shardingsphere-infra/shardingsphere-infra-executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/federate/FederateJDBCExecutorTest.java

```diff
@@ -0,0 +1,169 @@
+            + "WHERE t_order_federate.user_id = t_user_info.user_id";
+    
+    private ShardingSphereOptimizer optimizer;
+    
+    @Before
+    public void init() throws Exception {
+        String schemaName = "federate_jdbc_0";
+        Map<String, List<String>> columnMap = initializeColumnMap();
+        Map<String, List<String>> tableMap = initializeTableMap();
+        OptimizeContextFactory optimizeContextFactory = intializeOptimizeContextFactory(schemaName, tableMap);
+        FederateLogicSchema calciteSchema = intializeCalciteSchema(schemaName, columnMap, tableMap);
+        OptimizeContext context = optimizeContextFactory.create(schemaName, calciteSchema);
+        optimizer = new ShardingSphereOptimizer(context);
+    }
+    
+    @Test
+    public void testSimpleSelect() {
+        RelNode relNode = optimizer.optimize(SELECT_SQL_BY_ID_ACROSS_SINGLE_AND_SHARDING_TABLES);
+        String temp = "EnumerableCalc(expr#0..4=[{inputs}],proj#0..1=[{exprs}],information=[$t4])"
+            + "  EnumerableCalc(expr#0..4=[{inputs}],expr#5=[=($t1,$t3)],proj#0..4=[{exprs}],$condition=[$t5])"
+            + "    EnumerableNestedLoopJoin(condition=[true],joinType=[inner])"
+            + "      EnumerableTableScan(table=[[federate_jdbc_0,t_order_federate]])"
+            + "      EnumerableTableScan(table=[[federate_jdbc_0,t_user_info]])";
+        String expected = temp.replaceAll("\\s*", "");
+        String actual = relNode.explain().replaceAll("\\s*", "");
+        assertEquals(expected, actual);
+    }
+    
+    private Map<String, List<String>> initializeTableMap() {
+        Map<String, List<String>> result = new HashMap<>();
+        List<String> tableList = new ArrayList<>();
+        tableList.add("t_order_federate");
+        tableList.add("t_user_info");
+        result.put("federate_jdbc_0", tableList);
+        return result;
+    }
+
+    private Map<String, List<String>> initializeColumnMap() {
+        final Map<String, List<String>> result = new HashMap<>();
+        List<String> columnList = new ArrayList<>();
+        columnList.add("order_id");
+        columnList.add("user_id");
+        columnList.add("status");
+        result.put("t_order_federate", columnList);
+        List<String> columnList2 = new ArrayList<>();
+        columnList2.add("user_id");
+        columnList2.add("information");
+        result.put("t_user_info", columnList2);
+        return result;
+    }
+    
+    private FederateLogicSchema intializeCalciteSchema(final String schemaName, final Map<String, List<String>> columnMap, final Map<String, List<String>> tableMap) {
+        FederateSchemaMetadata federateSchemaMetadata = buildSchemaMetaData(schemaName, tableMap.get(schemaName), columnMap);
+        return new FederateLogicSchema(federateSchemaMetadata, null);
+    }
+    
+    private FederateSchemaMetadata buildSchemaMetaData(final String schemaName, final List<String> tableNames, final Map<String, List<String>> tableColumns) {
```

Comment:
Could we use the constructor of `FederateSchemaMetadatas` to generate `FederateSchemaMetadata` automatically?

Reply:
I am afraid we can not do this. As FederateSchemaMetadatas have common attributes with optimizeContextFactory.
![image](https://user-images.githubusercontent.com/22066046/126067434-9b939dd8-12ad-486e-ae5e-7a5e51e67f09.png)